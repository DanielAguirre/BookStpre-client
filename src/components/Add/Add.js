import React from 'react';
import CreateBook from '../CreateBook';

const Add = () => (
  <div className="form">
    <h3>Add a new Book to the Catalog</h3>
    <CreateBook/>
  </div>
);

module.exports = Add;