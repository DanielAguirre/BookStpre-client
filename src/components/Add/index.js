import React from 'react';
import CreateBook from '../CreateBook';
import glamorous from 'glamorous'

const Div = glamorous.div({
  display: 'flex',
  width: '80%',
  margin: '0 auto',
  fontSize: '16px',
  flexWrap: 'wrap'
});

const H3 = glamorous.h3({
  width: '100%',
  margin: '0 auto',
});

const Add = ({edit, params={}}) => {
  return (<Div className="form">
    <H3>{
      edit? 'Edit the Book'
      :'Add a new Book to the Catalog' }</H3>
    <CreateBook params={params} id={params.id}/>
  </Div>
)};

module.exports = Add;