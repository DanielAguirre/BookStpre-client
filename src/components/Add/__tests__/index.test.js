import React from 'react';
import { shallow } from 'enzyme';
import Add from '../index';

it('Render the add componet', () => {
  const componet = shallow(<Add/>);
  expect(componet).toMatchSnapshot();
});

it('Render the add componet to edit a book', () => {
  const componet = shallow(<Add edit={true} params={{id:1}} />);
  expect(componet).toMatchSnapshot();
});