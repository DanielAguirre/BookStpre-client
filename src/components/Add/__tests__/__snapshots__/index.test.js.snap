exports[`test Render the add componet 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <glamorous(div)
    className="form">
    <glamorous(h3)>
        Add a new Book to the Catalog
    </glamorous(h3)>
    <CreateBook
        id={undefined}
        params={Object {}} />
</glamorous(div)>,
  "nodes": Array [
    <glamorous(div)
      className="form">
      <glamorous(h3)>
            Add a new Book to the Catalog
      </glamorous(h3)>
      <CreateBook
            id={undefined}
            params={Object {}} />
</glamorous(div)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Add />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <glamorous(div)
          className="form">
          <glamorous(h3)>
                    Add a new Book to the Catalog
          </glamorous(h3)>
          <CreateBook
                    id={undefined}
                    params={Object {}} />
</glamorous(div)>,
        "_debugID": 2,
        "_renderedOutput": <glamorous(div)
          className="form">
          <glamorous(h3)>
                    Add a new Book to the Catalog
          </glamorous(h3)>
          <CreateBook
                    id={undefined}
                    params={Object {}} />
</glamorous(div)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Add />,
}
`;

exports[`test Render the add componet to edit a book 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <glamorous(div)
    className="form">
    <glamorous(h3)>
        Edit the Book
    </glamorous(h3)>
    <CreateBook
        id={1}
        params={
            Object {
                "id": 1,
              }
        } />
</glamorous(div)>,
  "nodes": Array [
    <glamorous(div)
      className="form">
      <glamorous(h3)>
            Edit the Book
      </glamorous(h3)>
      <CreateBook
            id={1}
            params={
                  Object {
                        "id": 1,
                      }
            } />
</glamorous(div)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Add
        edit={true}
        params={
                Object {
                        "id": 1,
                      }
        } />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "edit": true,
          "params": Object {
            "id": 1,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <glamorous(div)
          className="form">
          <glamorous(h3)>
                    Edit the Book
          </glamorous(h3)>
          <CreateBook
                    id={1}
                    params={
                              Object {
                                        "id": 1,
                                      }
                    } />
</glamorous(div)>,
        "_debugID": 4,
        "_renderedOutput": <glamorous(div)
          className="form">
          <glamorous(h3)>
                    Edit the Book
          </glamorous(h3)>
          <CreateBook
                    id={1}
                    params={
                              Object {
                                        "id": 1,
                                      }
                    } />
</glamorous(div)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Add
    edit={true}
    params={
        Object {
            "id": 1,
          }
    } />,
}
`;
